app/views/layout.erb

```diff
@@ -0,0 +1,46 @@
+<!doctype html>
+<html>
+  <head>
+    <title>Educational Feedback</title>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css">
+
+  </head>
+  <body>
+
+    <div class="container">
+
+      <div id="navbar" class="light-fade">
+           <div class="wrapper">
+             <nav>
+               <% if authenticate_teacher! %>
```

Comment:
this authenticate_teacher method is meant to redirect and shouldn't be a helper method. You should be checking if current_user is a teacher or student here instead.